src/component/EmailStarred/index.js

```diff
@@ -0,0 +1,135 @@
+import { useState } from "react";
+import EmailItem from "../EmailItem";
+import styles from "./EmailStarred.module.scss";
+import ChevronRightIcon from "@mui/icons-material/ChevronRight";
+import ChevronLeftIcon from "@mui/icons-material/ChevronLeft";
+import clsx from "clsx";
+import { useEffect } from "react";
+import { API_URL } from "../../constants";
+
+const token = localStorage.getItem("token");
+
+function EmailStarred() {
+  const [checkedItems, setCheckedItems] = useState([]);
+  const [checkedStar, setCheckedStar] = useState([]);
+  const [emailList, setEmailList] = useState([]);
+  const [data, setData] = useState({});
+  const [page, setPage] = useState(1);
+  useEffect(() => {
+    fetch(`${API_URL}/get-starred`, {
+      headers: { Authorization: `Bearer ${localStorage.getItem("token")}` },
+    })
+      .then((response) => {
+        if (response.ok) {
+          return response.json();
+        }
+      })
+      .then((data) => {
+        setEmailList(data.result.data.data);
+        setData(data.result.data);
+      });
+  }, [setEmailList]);
+  const handleCheckAll = () => {
+    if (emailList.length === checkedItems.length) {
+      setCheckedItems([]);
+      return;
+    }
+    setCheckedItems(emailList.map((item) => item.email.id));
+  };
+  const handleNext = () => {
+    if (emailList.length + (page - 1) * 50 < data.total) {
+      setPage(page + 1);
+    }
+  };
+  const handleBack = () => {
+    if (page !== 1) {
+      setPage(page - 1);
+    }
+  };
+  const handleStar = async (index) => {
+    const newEmail = { ...emailList[index] };
```

Comment:
No need clone, because you don't modify it, just `const email = emailList[index]`